src/parser/ast/ArrowFunctionExpressionNode.h

```diff
@@ -73,6 +73,9 @@ class ArrowFunctionExpressionNode : public ExpressionNode {
     virtual void generateExpressionByteCode(ByteCodeBlock* codeBlock, ByteCodeGenerateContext* context, ByteCodeRegisterIndex dstIndex)
     {
         CodeBlock* blk = context->m_codeBlock->asInterpretedCodeBlock()->childBlocks()[m_subCodeBlockIndex];
+        if (blk->usesArgumentsObject()) {
+            codeBlock->pushCode(EnsureArgumentsObject(ByteCodeLOC(m_loc.index)), context, this);
+        }
```

Comment:
I want to check that in the case of nested arrow functions where the innermost arrow function uses arguments, is `ArgumentsObject` only allocated on the outermost normal function's env?

Reply:
```js
function ff() {
    var a = (k) => { return () => { return arguments[0] + 1;}};
    return a(1)();
}

```

Reply:
I mark useArgumentsObject flag from innermost to arguments object holder.
you can see how I mark this things on ScriptParser::generateCodeBlockTreeFromASTWalker(Context... here

Reply:
Yes, I checked that `m_usesArgumentsObject` flag is marked for every inner arrow functions. But for the above sample code, it seems that `ArgumentsObject` is created twice, one for the outermost `ff` function and the other for the first inner arrow function which is not necessary. Of course, this does not trigger any error but generating ArgumentsObject in arrow function is not a correct operation. Would you please check once more?

Reply:
okay it seems wrong that way. I will check it tomorrow

Reply:
I fixed it